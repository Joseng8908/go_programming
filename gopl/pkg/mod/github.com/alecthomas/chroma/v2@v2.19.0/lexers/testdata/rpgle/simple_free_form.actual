**free

dcl-pi *n;
    n int(10) const;
end-pi;

dcl-s myvar varchar(52) inz('hello world') ccsid(*utf8);

myvar += %char(n);

dsply myvar;
return;