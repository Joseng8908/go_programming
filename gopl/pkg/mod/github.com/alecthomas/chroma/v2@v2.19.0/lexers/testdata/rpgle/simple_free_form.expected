[
  {"type":"CommentSpecial","value":"**free"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Keyword","value":"dcl-pi"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Literal","value":"*n"},
  {"type":"Punctuation","value":";"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"Text","value":"n"},
  {"type":"TextWhitespace","value":" "},
  {"type":"KeywordType","value":"int"},
  {"type":"Punctuation","value":"("},
  {"type":"LiteralNumber","value":"10"},
  {"type":"Punctuation","value":")"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"const"},
  {"type":"Punctuation","value":";"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Keyword","value":"end-pi"},
  {"type":"Punctuation","value":";"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Keyword","value":"dcl-s"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"myvar"},
  {"type":"TextWhitespace","value":" "},
  {"type":"KeywordType","value":"varchar"},
  {"type":"Punctuation","value":"("},
  {"type":"LiteralNumber","value":"52"},
  {"type":"Punctuation","value":")"},
  {"type":"TextWhitespace","value":" "},
  {"type":"KeywordReserved","value":"inz"},
  {"type":"Punctuation","value":"("},
  {"type":"LiteralString","value":"'hello world'"},
  {"type":"Punctuation","value":")"},
  {"type":"TextWhitespace","value":" "},
  {"type":"KeywordReserved","value":"ccsid"},
  {"type":"Punctuation","value":"("},
  {"type":"Literal","value":"*utf8"},
  {"type":"Punctuation","value":");"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Text","value":"myvar"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"+="},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameFunctionMagic","value":"%char"},
  {"type":"Punctuation","value":"("},
  {"type":"Text","value":"n"},
  {"type":"Punctuation","value":");"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Keyword","value":"dsply"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"myvar"},
  {"type":"Punctuation","value":";"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Keyword","value":"return"},
  {"type":"Punctuation","value":";"}
]
